java/garbage-collector/src/test/java/sleeper/garbagecollector/GarbageCollectorIT.java

```diff
@@ -141,27 +145,31 @@ void shouldNotCollectFileMarkedAsActive() throws Exception {
 
             // Then
             assertThat(Files.exists(filePath)).isTrue();
-            assertThat(getFilesInReadyForGCTable(instanceProperties, tableProperties)).isEmpty();
+            assertThat(stateStore.getAllFileReferences().getFiles())
+                    .flatMap(FileReferences::getFilename)
+                    .containsExactly(filePath.toString());
         }
 
         @Test
-        void shouldNotCollectFileMarkedAsReadyForGCBeforeSpecifiedDelay() throws Exception {
+        void shouldNotCollectFileWithNoReferencesBeforeSpecifiedDelay() throws Exception {
             // Given
             instanceProperties = createInstanceProperties();
             tableProperties = createTable(TEST_TABLE_NAME, instanceProperties);
             Instant currentTime = Instant.parse("2023-06-28T13:46:00Z");
             StateStore stateStore = setupStateStoreAndFixTime(currentTime);
-            java.nio.file.Path filePath = tempDir.resolve("test-file.parquet");
-            createReadyForGCFile(filePath.toString(), stateStore);
+            java.nio.file.Path oldFile = tempDir.resolve("old-file.parquet");
+            java.nio.file.Path newFile = tempDir.resolve("new-file.parquet");
+            createFileWithNoReferencesByCompaction(stateStore, currentTime, oldFile, newFile);
 
             // When
+            stateStore.fixTime(currentTime);
             createGarbageCollector(instanceProperties, stateStoreProvider).run();
```

Comment:
A lot of these tests aren't really valid because it's not controlling the time used inside the garbage collector. The assertion on this test in particular has been broken.